x-pack/plugins/triggers_actions_ui/public/common/index_controls/index.ts

```diff
@@ -8,48 +8,30 @@
 import { uniq } from 'lodash';
 import { HttpSetup } from 'kibana/public';
 import { i18n } from '@kbn/i18n';
-import {
-  loadIndexPatterns,
-  getMatchingIndices,
-  getESIndexFields,
-  getSavedObjectsClient,
-} from '../lib/data_apis';
+import { loadIndexPatterns, getMatchingIndices, getESIndexFields } from '../lib/data_apis';
 
 export interface IOption {
   label: string;
   options: Array<{ value: string; label: string }>;
 }
 
-export const getIndexPatterns = async () => {
-  // TODO: Implement a possibility to retrive index patterns different way to be able to expose this in consumer plugins
-  if (getSavedObjectsClient()) {
-    const indexPatternObjects = await loadIndexPatterns();
-    return indexPatternObjects.map((indexPattern: any) => indexPattern.attributes.title);
-  }
-  return [];
-};
-
-export const getIndexOptions = async (
-  http: HttpSetup,
-  pattern: string,
-  indexPatternsParam: string[]
-) => {
+export const getIndexOptions = async (http: HttpSetup, pattern: string) => {
   const options: IOption[] = [];
 
   if (!pattern) {
     return options;
   }
 
-  const matchingIndices = (await getMatchingIndices({
-    pattern,
-    http,
-  })) as string[];
-  const matchingIndexPatterns = indexPatternsParam.filter((anIndexPattern) => {
-    return anIndexPattern.includes(pattern);
-  }) as string[];
+  const [matchingIndices, matchingIndexPatterns] = await Promise.all([
+    getMatchingIndices({
+      pattern,
+      http,
+    }),
+    loadIndexPatterns(pattern),
+  ]);
```

Comment:
Fetch both the indices and the index patterns, as filtered by the search keyword (pattern)